src/Storage/TSStorageManager.m

```diff
@@ -287,6 +287,17 @@ - (BOOL)databasePasswordAccessible
     return NO;
 }
 
+- (void)backgroundedAppDatabasePasswordInaccessibleWithError:(NSError *)error
+{
+    OWSAssert([UIApplication sharedApplication].applicationState == UIApplicationStateBackground);
+
+    // Presumably this happened in response to a push notification. It's possible that the keychain is corrupted
+    // but it could also just be that the user hasn't yet unlocked their device since our password is
+    // kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    [NSException raise:TSStorageManagerExceptionNameDatabasePasswordInaccessibleWhileBackgrounded
+                format:@"Unable to access database password. No unlock since device restart? Error: %@", error];
```

Comment:
Let's:

* Write error.description to the logs.
* Write [UIApplication sharedApplication].applicationState to the logs.
* Flush the logs.